packages/react-app/src/hooks/useStaticJsonRPC.js

```diff
@@ -2,32 +2,75 @@ import { useCallback, useEffect, useState } from "react";
 import { ethers } from "ethers";
```

Comment:
I think we can make the changes on this file a bit more straight-forward.

We can leave as it was before and only adding a `localProvider = null` param to the `useStaticJsonRPC` hook. Then, inside the hook:

- We check if localProvider exists and is mainnet. 
- If is mainnet => init the provider useState with localProvider.
- If is NOT mainnet => init useState with null (as it was)
- On the useEffect, we don't call handleProviders if localProvider exists and is mainnet. 


This way the hook will return localProvider if mainnet, and if not it'll initialize a new provider.

Reply:
sure let me think on this